interview-questions.md

```diff
@@ -7,90 +7,89 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. Why would you use the method `super()`?
 
-  Your answer:
+  Your answer: super() is used for a new constructor to call on a different class and gain access to that classes constructors.
 
-  Researched answer:
+  Researched answer: Calls the constructor from the parent class and gives us access to the parent classes properties and methods
 
 
 
 2. What is the DOM? What is the virtual DOM?
 
-  Your answer:
+  Your answer: Not sure what it stands for, but I think it basically represents an object on an html page as well as its javascript and css. 
 
-  Researched answer:
+  Researched answer: Stands for Document Object Model. It is an interface that allows functions to dynamically change anything on a document/webpage.
 
 
 
 3. What is a higher-order function?
 
-  Your answer:
+  Your answer: It's different from a built in function, higher-order functions can be attached to an object or another method and follows a set syntax inside its parenthesis that will return the outcome with your own twist.  
 
-  Researched answer:
+  Researched answer: They take functions that take other functions as arguments. They can also return functions as return values, or both.
 
 
 
 4. What is JSX?
 
-  Your answer:
+  Your answer: A combination of javascript and html so that you don't need 2 separate files.
 
-  Researched answer:
+  Researched answer: Stands for JavaScript Extensible Markup Language. It allows and makes it easier to add and write HTML in React.
 
 
+5. What is React? 
 
-5. What is React?
+  Your answer: A javascript library that allows users to add dynamic components inside a javascript file while combining javascript and html together.
 
-  Your answer:
-
-  Researched answer:
+  Researched answer: A JavaScript library for building UI. Let's you create dynamic components on a document.
 
 
 
 6. What is yarn? What file(s) are updated when you run `$ yarn`?
 
-  Your answer:
+  Your answer: yarn allows us to run a .js file like a webpage. As we change anything inside that file, the yarn page also updates our changes immediately. yarn adds a .yarn file and a .json file to your directory. 
```

Comment:
Just about! It would probably be closer to saying that its a package manager allowing us to manage dependencies and run some applications. Upon running the command, it would work with its 'yarn.lock' file and update the 'package.json' file, as the .yarn and.json could be any random file, and the json file extension isnt specific to yarn.